File: integration-tests/hibernate-reactive-panache/src/test/java/io/quarkus/it/panache/reactive/TestReactiveTransactionTest.java
package io.quarkus.it.panache.reactive;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import io.quarkus.hibernate.reactive.panache.Panache;
import io.quarkus.test.TestReactiveTransaction;
import io.quarkus.test.junit.QuarkusTest;
import io.quarkus.test.vertx.UniAsserter;

@QuarkusTest
@TestReactiveTransaction
public class TestReactiveTransactionTest {

    @Test
    public void testTestTransaction(UniAsserter asserter) {
        asserter.assertNotNull(() -> Panache.currentTransaction());
    }

    @BeforeEach
    public void beforeEach(UniAsserter asserter) {
        asserter.assertNotNull(() -> Panache.currentTransaction());
    }
}

File: integration-tests/hibernate-reactive-panache-kotlin/src/test/kotlin/io/quarkus/it/panache/kotlin/TestReactiveTransactionTest.kt
package io.quarkus.it.panache.kotlin

import io.quarkus.it.panache.reactive.kotlin.Person
import io.quarkus.test.TestReactiveTransaction
import io.quarkus.test.junit.QuarkusTest
import io.quarkus.test.junit.vertx.UniAsserter
import org.hibernate.reactive.mutiny.Mutiny
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

@QuarkusTest
class TestReactiveTransactionTest {

    @TestReactiveTransaction
    @Test
    fun testTestTransaction(asserter: UniAsserter) {
        asserter.assertNotNull { Person.currentTransaction() }
    }

    @TestReactiveTransaction
    @BeforeEach
    fun beforeEach(asserter: UniAsserter) {
        asserter.assertNotNull<Mutiny.Transaction> { io.quarkus.hibernate.reactive.panache.Panache.currentTransaction() }
    }
}